catalog/rocks/validate.go

```diff
@@ -105,12 +108,29 @@ func ValidateTagID(v interface{}) error {
 	if !ok {
 		panic(ErrInvalidType)
 	}
-	if len(s) == 0 {
+	// http://git-scm.com/docs/git-check-ref-format
+	tag := string(s)
+	if len(tag) == 0 {
 		return ErrRequiredValue
 	}
-	if !reValidBranchID.MatchString(string(s)) {
+	if tag == "@" {
+		return ErrInvalidValue
+	}
+	if strings.HasSuffix(tag, ".") || strings.HasSuffix(tag, ".lock") || strings.HasSuffix(tag, "/") {
+		return ErrInvalidValue
+	}
+	if strings.Contains(tag, "..") || strings.Contains(tag, "//") || strings.Contains(tag, "@{") {
+		return ErrInvalidValue
+	}
+	// Unlike git, we do allow '~'.  That supports migration from our previous ref format where commits started with a tilde.
+	if strings.ContainsAny(tag, "^:?*[\\") {
 		return ErrInvalidValue
 	}
+	for _, r := range tag {
+		if r <= minControlCharCode {
```

Comment:
`minControlCode` is misnamed as space is not a control code.  Perhaps replace the constant with a named function,
```go
func isControlCodeOrSpace(r rune) {
	return r <= 0x20
}
```
(Octal hasn't made sense since PDP-8)

Reply:
cool - took it.